client/state/billing-transactions/transaction-filters/reducer.js

```diff
@@ -0,0 +1,71 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import {
+	BILLING_TRANSACTIONS_FILTER_SET_APP,
+	BILLING_TRANSACTIONS_FILTER_SET_DATE,
+	BILLING_TRANSACTIONS_FILTER_SET_PAGE,
+	BILLING_TRANSACTIONS_FILTER_SET_QUERY,
+} from 'state/action-types';
+import { combineReducers, createReducer, keyedReducer } from 'state/utils';
+
+/**
+ * Returns the updated app filter state after an action has been dispatched
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {String}        Updated state
+ */
+export const app = createReducer( '', {
+	[ BILLING_TRANSACTIONS_FILTER_SET_APP ]: ( state, action ) => action.app,
+} );
+
+/**
+ * Returns the updated date filter state after an action has been dispatched
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const date = createReducer(
+	{ newest: true },
+	{
+		[ BILLING_TRANSACTIONS_FILTER_SET_DATE ]: ( state, { newest, month, before } ) => ( {
+			newest,
+			month,
+			before,
```

Comment:
Feels like we're adding extra overhead by storing several types of dates here (`month` and `before`), when all of them are to be applied to the same filter. This could easily put the state out of sync - what will happen if we set dates both to `month` and `before`? I'd probably try to use a single field for storing the date; and possibly another field (that would specify where to look, relative to the date passed - `=` or `<`). What do you think?